feat: convert mdx components in web app to mdx-components package

jharvey10 opened the pull request:
Closes #1103
Closes #1384 
Closes #1413 
Closes #1496 

### Things to check

- [x] Did we remember to remove the components from components.js?
- [x] Did we remember to add each component to `mdx-components/src/main/index.ts`?
- [x] Did we remember to add the styles for each component to `mdx-components/src/main/index.scss`?
- [x] Did we remember to move over the stories files?
- [x] Did we remove all components that are re-exports of carbon ones in favor if the core carbon version?
- [x] Did we remember to update the unit test snapshots?
- [x] Did we remember to export the typescript props interface from each component (and export it from index.ts)?

### Open questions

- [x] What do we do about the `pages/mdx` files. Should they be roped into the storybook stories in the mdx-components package? (#1499)

### Testing / reviewing

Compare the component visuals and functionality to what they were before the conversion.

alisonjoseph committed:
feat(mdx-components): add caption component

alisonjoseph committed:
feat(web-app): remove caption mdx component

alisonjoseph committed:
feat(web-app): remove caption

alisonjoseph committed:
feat(mdx-components): add accordion component

jharvey10 committed:
feat(mdx-components): add global MdxComponent helper interface

alisonjoseph committed:
feat(web-app): remove accordion component

alisonjoseph committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of https://github.com/carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

alisonjoseph committed:
feat(mdx-components): update accordion

jharvey10 committed:
fix(mdx-components): update caption class name, add some docs

alisonjoseph committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of https://github.com/carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

jharvey10 committed:
fix(mdx-components): add optional types to accordion className prop

jharvey10 committed:
fix(mdx-components): adjust packaging includes/excludes

jharvey10 committed:
fix(mdx-components): correct typings of MdxComponent

jharvey10 committed:
fix(mdx-components): re-export carbon accordion item

jharvey10 committed:
feat(mdx-components): convert video component

mattrosno commented:
@jdharvey-ibm @alisonjoseph we shouldn't need this page, once all of these MDX components are documented in individual Storybook stories.

https://next.carbondesignsystem.com/mdx

alisonjoseph commented:
@mattrosno @jdharvey-ibm Let's wait until everything looks good and is approved before removing the test mdx pages, so we can review things in the site easily and compare to live.

alisonjoseph committed:
feat(mdx-components): add aside component

alisonjoseph committed:
feat(web-app): remove aside component

alisonjoseph committed:
feat(mdx-components): add card-group

alisonjoseph committed:
feat(web-app): remove card-group

alisonjoseph commented on packages/mdx-components/src/main/card-group/card-group.tsx:
```diff
@@ -0,0 +1,53 @@
+
+// TODO: ask Joe - how can we use this in here?
+// import { mediaQueries, useMatchMedia }
+// from '../../../../services/web-app/utils/use-match-media.js'
```
@jdharvey-ibm tagging you now so I don't forget to ask you next week

jdharvey-ibm replied:
@alisonjoseph There's no easy way for us to use this in both places :/

I think core Carbon does have a media query hook that we could get at. That could potentially be used here instead.

If not, our best option is to copy-paste those two imports from the web-app into the mdx-components package.

alisonjoseph committed:
feat(mdx-components): add divider

alisonjoseph committed:
feat(web-app): remove divider from web app

alisonjoseph committed:
feat(mdx-components): add do dont

alisonjoseph committed:
feat(web-app): remove do dont

francinelucca committed:
feat(mdx-components): migrate GifPlayer component into mdx-components

Closes #1103

alisonjoseph committed:
feat(mdx-components): add grid

alisonjoseph committed:
feat(web-app): remove grid transform

alisonjoseph committed:
fix(mdx-components): accordion size prop

alisonjoseph committed:
wip: comment out grid

alisonjoseph committed:
fix(mdx-components): update accordion styles

jharvey10 committed:
build: scope nextjs eslint config to web-app

alisonjoseph committed:
feat(mdx-components): add use match media

alisonjoseph committed:
wip: fix grid

jharvey10 committed:
refactor(mdx-components): move media query functions to utils module

jharvey10 committed:
docs(mdx-components): fix typo

francinelucca committed:
feat(mdx-components): migrate paragraph and pageDescription components

Closes #1103

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
feat(mdx-components): migrate Profile component

Closes #1103

jharvey10 committed:
feat(mdx-components): convert title

jharvey10 committed:
feat(web-app): remove title component

jharvey10 committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

alisonjoseph committed:
feat(mdx-components): add art direction

alisonjoseph committed:
feat(web-app): remove art direction

alisonjoseph committed:
fix: imports

alisonjoseph committed:
feat(mdx-components): add column

alisonjoseph committed:
feat(mdx-components): add row

francinelucca committed:
feat(mdx-components): migrate resource card

Closes #1103

alisonjoseph committed:
fix: col and row errors

francinelucca committed:
feat(mdx-components): migrate image wrapper component

Closes #1103

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
feat(mdx-components): migrate MiniCard

Co-authored-by: Andrea N. Cardona <andreancardona@gmail.com>

andreancardona committed:
feat: migrate article card

andreancardona committed:
feat: inline notification mdx

francinelucca committed:
fix(mdx-components): nest gifPlayer classes

Closes #1103

alisonjoseph committed:
feat(mdx-components): group cards

alisonjoseph committed:
fix(mdx-components): column component missing props

alisonjoseph committed:
fix(mdx-components): column component missing props

alisonjoseph committed:
feat(mdx-components): add storybook demo

alisonjoseph committed:
feat(web-app): remove storybook demo

alisonjoseph commented on packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx:
```diff
@@ -0,0 +1,178 @@
+import { MdxComponent } from '../interfaces.js'
+import { withPrefix } from '../utils.js'
+
+/* eslint-disable  @typescript-eslint/no-explicit-any  -- TODO ask Joe how to fix this */
+type Variants = { [key: string]: any }
+/* eslint-enable @typescript-eslint/no-explicit-any  -- Re-enable rule */
+
+interface StorybookDemoProps {
```
@jdharvey-ibm can you take a look at this file when you get a chance 😅

alisonjoseph commented on packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx:
```diff
@@ -0,0 +1,178 @@
+  }
+
+  const multipleVariants = variants?.length > 1
+  console.log(' 🐬 ~ variants', variants)
```
```suggestion
```

alisonjoseph committed:
Update packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx

alisonjoseph committed:
feat(mdx-components): add card story doc improvements

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
feat(mdx-components): migrate headings markdown comps

Closes #1103

andreancardona committed:
feat: preview component

andreancardona committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of https://github.com/carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

alisonjoseph committed:
wip: merge

alisonjoseph committed:
wip: merge

francinelucca committed:
feat(mdx-components): migrate blockquote

Closes #1103

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
fix(web-app): import headings from mdx-components

Closes #1103

francinelucca committed:
fix(mdx-components): add mdx guidance to headings SB

Closes #1103

francinelucca committed:
feat(mdx-components): migrate PageTable component

Closes #1103

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
fix(web-app): pull paragraph and pagedescription from mdx-components pkg

Closes #1103

jharvey10 committed:
fix(mdx-components): add typing to storybook-demo variants prop

jharvey10 committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

andreancardona committed:
feat: ul ol and li markdown

andreancardona committed:
fix: merge conflict

alisonjoseph committed:
feat(mdx-components): add typeweight

alisonjoseph committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of https://github.com/carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

alisonjoseph committed:
feat: organize storybook

alisonjoseph committed:
feat: organize storybook

francinelucca committed:
feat(mdx-components): add Link component

Closes #1103

francinelucca committed:
fix(web-app): pull Link component from mdx-components package

Closes #1103

alisonjoseph committed:
feat: organize storybook

jharvey10 committed:
test(mdx-components): fix ts-node file loading in ava tests

jharvey10 committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

alisonjoseph committed:
Merge branch 'main' into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

jharvey10 committed:
feat(mdx-components): add new NonScalarNode utility type

alisonjoseph committed:
feat(mdx-components): add noanchor prop to h2

alisonjoseph committed:
feat(web-app): component cleanup

alisonjoseph committed:
feat(mdx-components): add typescaletable

alisonjoseph committed:
feat(web-app): remove typescaletable

jharvey10 committed:
feat(mdx-components): convert tabs component

jharvey10 committed:
feat(web-app): remove tabs component

jharvey10 committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
feat(mdx-components): add Code component

Closes #1103

francinelucca committed:
fix(web-app): use Code component from mdx-components pckg

Closes #1103

francinelucca committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

andreancardona committed:
feat: color grid and color palette convert

Co-authored-by:  Joe Harvey <jdharvey-ibm@users.noreply.github.com>
Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>

andreancardona committed:
Merge branch '1103-convert-mdx-components-in-web-app-to-mdx-components-package' of https://github.com/carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

andreancardona committed:
feat: status indicator mdx

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into 1103-convert-mdx-components-in-web-app-to-mdx-components-package

francinelucca committed:
fix(mdx-components): remove one-off components migrations

francinelucca committed:
fix(web-app): reintroduce one-off components

francinelucca committed:
fix(web-app): reintroduce deleted stories

francinelucca committed:
fix(mdx-components): fix broken build and minor sb media fixes

francinelucca committed:
fix(web-app): export colorblock from mdx components

francinelucca committed:
fix(mdx-components): add test for migrated comps, fix some props

Co-authored-by: Joe Harvey <jdharvey-ibm@users.noreply.github.com>

francinelucca committed:
fix(web-app): export one-off carbon components from web app

francinelucca committed:
fix(web-app): incorrect import

francinelucca committed:
fix(mdx-components): more test coverage

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

jharvey10 committed:
deps(mdx-components): hoist @carbon/react to base workspace

jharvey10 committed:
test(mdx-components): add more unit tests

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>

francinelucca committed:
fix(mdx-components): address some proptype warnings

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

francinelucca commented:
we removed most prop warnings @jdharvey-ibm and the ones left behind are reasonable!

jharvey10 committed:
test(mdx-components): update snapshots

jharvey10 committed:
fix(mdx-components): visual consistency fixes and remove some warnings

Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>

jharvey10 committed:
fix(mdx-components): visual component fixes

- fix title to use h4 without autolink (use p instead of an h4)
- adjust style of some components as siblings of headers
- add new "header" utility class to h1-h4

Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>

jharvey10 committed:
test(mdx-components): update unit test snapshots

jharvey10 committed:
fix(mdx-components): remove title styles from index.scss
